skrub/_table_vectorizer.py

```diff
@@ -112,6 +112,216 @@ def _check_transformer(transformer):
     return clone(transformer)
 
 
+def _get_preprocessors(*, cols, drop_null_fraction, n_jobs, add_tofloat32=True):
+    steps = [CheckInputDataFrame()]
+    transformers = [
+        CleanNullStrings(),
+        DropIfTooManyNulls(drop_null_fraction),
+        ToDatetime(),
+    ]
+    if add_tofloat32:
+        transformers.append(ToFloat32())
+    transformers += [
+        CleanCategories(),
+        ToStr(),
+    ]
+
+    for transformer in transformers:
+        steps.append(
+            wrap_transformer(
+                transformer,
+                cols,
+                allow_reject=True,
+                n_jobs=n_jobs,
+                columnwise=True,
+            )
+        )
+    return steps
+
+
+class SimpleCleaner(TransformerMixin, BaseEstimator):
+    """
+    A light transformer that preprocesses each column of a dataframe.
+
+    The ``SimpleCleaner`` detects dates that are represented as strings, and runs
+    some consistency checks on the column names. It also replaces null values
+    (e.g. NaN, N/A, None) with a null format suitable for the dtype of the column.
+    By default, columns that contain only null values are dropped. Differently
+    from the ``TableVectorizer``, the ``SimpleCleaner`` does not convert
+    numeric datatypes to float32, and thus does not ensure that null values have
+    dtype consistent across columns.
+
+    Parameters
+    ----------
+    drop_null_fraction : float or None, default=1.0
+        Fraction of null above which the column is dropped. If `drop_null_fraction`
+        is set to ``1.0``, the column is dropped if it contains only
+        nulls or NaNs (this is the default behavior). If `drop_null_fraction` is a
+        number in ``[0.0, 1.0)``, the column is dropped if the fraction of nulls
+        is strictly larger than `drop_null_fraction`. If `drop_null_fraction` is
+        ``None``, this selection is disabled: no columns are dropped based on the
+        number of null values they contain.
+
+    n_jobs : int, default=None
+        Number of jobs to run in parallel.
+        ``None`` means 1 unless in a joblib ``parallel_backend`` context.
+        ``-1`` means using all processors.
+
+    Attributes
+    ----------
+    all_processing_steps_ : dict
+        Maps the name of each column to a list of all the processing steps that were
+        applied to it.
+    Notes
+    -----
+    The ``SimpleCleaner`` performs the following set of transformations on each column:
+    - ``CleanNullStrings()``: replace strings used to represent null values
+    with actual null values.
+    - ``DropIfTooManyNulls()``: drop the column if it contains too many null values.
+    - ``ToDatetime()``: parse datetimes represented as strings and return them as
+    ``datetime64``.
```

Comment:
```suggestion
    actual datetimes with the correct dtype.
```

(they are called `Datetime` in polars)